modules/nw-customize-ingress-error-pages.adoc

```diff
@@ -0,0 +1,104 @@
+// Module included in the following assemblies:
+// * networking/ingress-controller-configuration.adoc
+
+[id="nw-customize-ingress-error-pages_{context}"]
+= Customizing HAProxy error code response pages
+
+As a cluster administrator, you can specify a custom error code response page for either 503, 404, or both error pages. The HAProxy router serves a 503 error page when the application pod is not running or a 404 error page when the requested URL does not exist. For example, if you customize the 503 error code response page, then the page is served when the application pod is not running, and the default 404 error code HTTP response page is served by the HAProxy router for an incorrect route or a non-existing route.
+
+Custom error code response pages are specified in a config map then patched to the Ingress Controller. The config map keys have two available file names as follows:
+`error-page-503.http` and `error-page-404.http`.
+
+Custom http error code response pages must follow the link:https://www.haproxy.com/documentation/hapee/latest/configuration/config-sections/http-errors/[HAProxy HTTP error page configuration guidelines]. Here is an example of the default {product-title} HAProxy router link:https://raw.githubusercontent.com/openshift/router/master/images/router/haproxy/conf/error-page-503.http[http 503 error code response page]. You can use the default content as a template for creating your own custom page.
+
+By default, the HAProxy router serves only a 503 error page when the application is not running or when the route is incorrect or non-existent. This default behavior is the same as the behavior on {product-title} 4.8 and earlier. If a config map for the customization of an HTTP error code response is not provided, and you are using a custom HTTP error code response page, the router serves a default 404 or 503 error code response page.
+
+[NOTE]
+====
+If you use the {product-title} default 503 error code page as a template for your customizations, the headers in the file require an editor than can use CRLF line endings.
+====
+
+.Procedure
+
+. Create a config map named `my-custom-error-code-pages` in the `openshift-config` namespace:
++
+[source,terminal]
+----
+$ oc -n openshift-config create configmap my-custom-error-code-pages \
+--from-file=error-page-503.http \
+--from-file=error-page-404.http
+----
+
+. Patch the Ingress Controller to reference the `my-custom-error-code-pages` config map by name:
++
+[source,terminal]
+----
+$ oc patch -n openshift-ingress ingresscontroller/default --patch \ '{"spec":{"httpErrorCodePages": "my-custom-error-code-pages"}}' --type=merge
+----
+
+. The Ingress Operator copies the `my-custom-error-code-pages` config map from the `openshift-config` namespace to the `openshift-ingress` namespace. A config map with name format `<your-ingresscontroller-name>-errorpages` gets created in the `openshift-ingress` namespace. Display the copy by running the following command:
```

Comment:
Please split this step by moving the result of the preceding step into a para and then start this step with the verb.

Perhaps like the foll, though I checked with peer review squad, and while preferred, we don't have a guideline on this:

```
. Patch the Ingress Controller to reference the `my-custom-error-code-pages` config map by name:
+
[source,terminal]
----
$ oc patch -n openshift-ingress ingresscontroller/default
----
+
The Ingress Operator copies the `my-custom-error-code-pages` config map from the `openshift-config` namespace to the `openshift-ingress` namespace. A config map with name format `<ingresscontroller_name>-errorpages` is created in the `openshift-ingress` namespace.

. Display the config map:
...
```
A few edits in there that I'd like to explain:
* In the replaceable values, use underscores according to the [code block](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#code-blocks-command-syntax-and-example-output) guidelines.  Ctrl+F underscore.
* s/gets created/is created/.  Even better would be active voice: "The Operator names the config map according to the pattern `<ingresscontroller_name>-errorpages` in the `openshift-ingress` namespace."
* My $0.02 is to omit the "by running the following command" for consistency with the other steps in this module (maybe the entire HTML page too.)